--- galternatives-0.13.5/galternatives.desktop.in~	2014-06-02 16:48:20.000000000 +0300
+++ galternatives-0.13.5/galternatives.desktop.in	2016-02-02 21:16:57.793585063 +0200
@@ -1,9 +1,8 @@
 [Desktop Entry]
-Version=1.0
 _Name=Alternatives Configurator
 _Comment=Configure the system default alternatives
 Type=Application
-Exec=/usr/sbin/galternatives
+Exec=/usr/bin/galternatives
 Icon=galternatives
 Terminal=false
 Categories=Settings;GTK;System;
